firebase-vscode/src/config-files.ts

```diff
@@ -0,0 +1,144 @@
+import * as fs from "fs";
+import * as path from "path";
+import * as vscode from "vscode";
+import { workspace } from "vscode";
+import { ExtensionBrokerImpl } from "./extension-broker";
+import { updateOptions, currentOptions } from "./options";
+import { RC } from "./rc";
+import { Config } from "./config";
+import { pluginLogger } from "./logger-wrapper";
+
+export function getRootFolders() {
+  if (!workspace) {
+    return [];
+  }
+  const folders = workspace.workspaceFolders
+    ? workspace.workspaceFolders.map((wf) => wf.uri.fsPath)
+    : [];
+  if (workspace.workspaceFile) {
+    folders.push(path.dirname(workspace.workspaceFile.fsPath));
+  }
+  return Array.from(new Set(folders));
+}
+
+function getConfigPath(): string {
+  // Usually there's only one root folder unless someone is using a
+  // multi-root VS Code workspace.
+  // https://code.visualstudio.com/docs/editor/multi-root-workspaces
+  // We are trying to play it safe by assigning the cwd
+  // based on where a .firebaserc or firebase.json was found but if
+  // the user hasn't run firebase init there won't be one, and without
+  // a cwd we won't know where to put it.
+  const rootFolders = getRootFolders();
+  for (const folder of rootFolders) {
+    if (fs.existsSync(path.join(folder, '.firebaserc'))
+      || fs.existsSync(path.join(folder, 'firebase.json'))) {
+      currentOptions.cwd = folder;
+      return folder;
+    }
+  }
+  currentOptions.cwd = rootFolders[0];
+  return rootFolders[0];
+}
+
+/**
+ * Parse firebase.json and .firebaserc from the configured location, if they
+ * exist, and write to memory.
+ */
+export function readFirebaseConfigs(context: vscode.ExtensionContext) {
+  const configPath = getConfigPath();
+  let firebaseRC: RC;
+  let firebaseJSON: Config;
+  try {
+    firebaseRC = RC.loadFile(path.join(configPath, '.firebaserc'));
+  } catch (e) {
+    if (e.message.includes('error trying to load')) {
+      firebaseRC = null;
+    } else {
+      pluginLogger.error(e.message);
+      throw e;
+    }
+  }
+  try {
+    firebaseJSON = Config.load({ configPath: path.join(configPath, 'firebase.json') });
+  }
+  catch (e) {
+    if (e.message.includes('could not locate')
+      || e.message.includes('Could not load config file')) {
```

Comment:
This error message matching seems a bit brittle - might be worth changing Config.load to return an error with status 404 and checking e.status instead